code/modules/mob/living/living.dm

```diff
@@ -648,66 +622,98 @@
-		resist_grab()
-		return
-
-	//unbuckling yourself
+	// only works if you're not cuffed.
+	if(!restrained(ignore_grab = TRUE) && pulledby)
+		var/old_gs = pulledby.grab_state
+		attempt_resist_grab(FALSE)
+		// Return as we should only resist one thing at a time. Give clickdelay if the grab wasn't passive.
+		return old_gs? TRUE : FALSE
+
+	// unbuckling yourself. stops the chain if you try it.
 	if(buckled && last_special <= world.time)
-		resist_buckle()
+		log_combat(src, buckled, "resisted buckle")
+		return resist_buckle()
 
-	// CIT CHANGE - climbing out of a gut
-	if(attempt_vr(src,"vore_process_resist",args)) return TRUE
+	// CIT CHANGE - climbing out of a gut.
+	if(attempt_vr(src,"vore_process_resist",args))
+		//Sure, give clickdelay for anti spam. shouldn't be combat voring anyways.
+		return TRUE
 
 	//Breaking out of a container (Locker, sleeper, cryo...)
-	else if(isobj(loc))
+	if(isobj(loc))
 		var/obj/C = loc
 		C.container_resist(src)
+		// This shouldn't give clickdelays sometime (e.g. going out of a mech/unwelded and unlocked locker/disposals bin/etc) but there's so many overrides that I am not going to bother right now.
+		return TRUE
 
-	else if(canmove)
+	if(CHECK_MOBILITY(src, MOBILITY_MOVE))
 		if(on_fire)
 			resist_fire() //stop, drop, and roll
-			return
-		if(resting) //cit change - allows resisting out of resting
-			resist_a_rest() // ditto
-			return
-		if(resist_embedded()) //Citadel Change for embedded removal memes
-			return
-		if(last_special <= world.time)
-			resist_restraints() //trying to remove cuffs.
-			return
-
+			// Give clickdelay
+			return TRUE
+	if(resting) //cit change - allows resisting out of resting
+		resist_a_rest() // ditto
+		// DO NOT GIVE CLCIKDELAY - resist_a_rest() handles spam prevention. Somewhat.
+		return FALSE
+	if(last_special <= world.time)
+		resist_restraints() //trying to remove cuffs.
+		// DO NOT GIVE CLICKDELAY - last_special handles this.
+		return FALSE
+	if(CHECK_MOBILITY(src, MOBILITY_USE) && resist_embedded()) //Citadel Change for embedded removal memes - requires being able to use items.
+		// DO NOT GIVE DEFAULT CLICKDELAY - This is a combat action.
+		changeNext_move(CLICK_CD_MELEE)
+		return FALSE
 
-/mob/proc/resist_grab(moving_resist)
-	return 1 //returning 0 means we successfully broke free
+/// Proc to resist a grab. moving_resist is TRUE if this began by someone attempting to move. Return FALSE if still grabbed/failed to break out. Use this instead of resist_grab() directly.
+/mob/proc/attempt_resist_grab(moving_resist, forced, log = TRUE)
+	if(!pulledby)	//not being grabbed
+		return TRUE
+	var/old_gs = pulledby.grab_state		//how strong the grab is
+	var/old_pulled = pulledby
+	var/success = do_resist_grab(moving_resist, forced)
+	if(log)
+		log_combat(src, old_pulled, "[success? "successfully broke free of" : "failed to resist"] a grab of strength [old_gs][moving_resist? " (moving)":""][forced? " (forced)":""]")
+	return success
+
+	/*!
```

Comment:
this comment being indented makes it look like it's supposed to be part of the previous proc